pkg/kubectl/cmd/portforward.go

```diff
@@ -17,21 +17,33 @@ limitations under the License.
 package cmd
 
 import (
+	"fmt"
 	"io"
 	"net/url"
 	"os"
 	"os/signal"
 
-	"github.com/golang/glog"
 	"github.com/renstrom/dedent"
 	"github.com/spf13/cobra"
 	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/client/restclient"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
 	"k8s.io/kubernetes/pkg/client/unversioned/portforward"
 	"k8s.io/kubernetes/pkg/client/unversioned/remotecommand"
 	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
 )
 
+type PortForwardOptions struct {
```

Comment:
This struct should only store flag values that are set from command line. `Config`, `Client`, `PortForwarder`, `StopChannel`, `ReadyChannel` can all be instantiated as local variables inside `ForwardPorts` without changing any params.